mongodb/intro/date-aggregation/aggregation-with-date-expression-operators-v.md

```diff
@@ -0,0 +1,119 @@
+---
+author: Stefan-Stojanovic
+
+aspects:
+  - introduction
+
+type: normal
+
+category: must-know
+
+links:
+- '[Date Operators](https://docs.mongodb.com/manual/reference/operator/aggregation/#date-expression-operators){documentation}'
+
+---
+# Aggregation With All Date Operators
+---
+## Content
+
+Now that we've seen all the date operators and know how to use them, let's take a look at an example of all operators being used in the same aggregation.
+
+**Note:** We are using the same document[1] as in the previous insights.
+
+Example with all operators:
+```js
+db.pokemon.aggregate([
+  { $match: { name: "Bulbasaur" } },
+  {
+    $project: {
+      dayOfTheYear: { $dayOfYear: "$createdAt" },
+      createdAtYear: { $year: "$createdAt" },
+      createdAtMonth: { $month: "$createdAt" },
+      dayOfTheMonth: { $dayOfMonth: "$createdAt" },
+      createdAtHour: { $hour: "$createdAt" },
+      createdAtMinute: { $minute: "$createdAt" },
+      createdAtSecond: { $second: "$createdAt" },
+      createdAtMillisecond: { $millisecond: "$createdAt" },
+      weekOfTheYear: { $week: "$createdAt" },
+      dayOfTheWeek: { $dayOfWeek: "$createdAt" },
+      _id: 0
+    }
+  }
+]);
+```
+
+Output:
+```js
+{ 
+  "dayOfTheYear": 282, 
+  "createdAtYear": 2019, 
+  "createdAtMonth": 10, 
+  "dayOfTheMonth": 9, 
+  "createdAtHour": 7, 
+  "createdAtMinute": 21, 
+  "createdAtSecond": 14, 
+  "createdAtMillisecond": 0, 
+  "weekOfTheYear": 40, 
+  "dayOfTheWeek": 4 
+}
+```
```

Comment:
can we use a more creative example? imo there is no benefit in this example because we've already shown this in all the previous insights. all that it does is add all the operators in the project stage

we could do something like finding the remaining time until one's birthday, the number of days remaining untill a contract renewal or something else. also, we don't have to include all the operators as long as we use an example that shows how a user could use these operators in a real-life application

edit: I notice that you give some examples in the first insight. why not make use of those? "These operators can be used to extract the specific time at which a document was made, they can be used to find a group of documents created on a specific date or to find all documents between 2 dates."